dotnet/src/Agents/Abstractions/AgentChat.cs

```diff
@@ -0,0 +1,184 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Agents.Extensions;
+using Microsoft.SemanticKernel.Agents.Internal;
+using Microsoft.SemanticKernel.ChatCompletion;
+
+namespace Microsoft.SemanticKernel.Agents;
+
+/// <summary>
+/// Point of interaction for one or more agents.
+/// </summary>
+public abstract class AgentChat
```

Comment:
The {de}serialization aspect of the agent chat/chat history needs to be captured and designed if not already.

Reply:
Totally agree.  I know we've discussed this on several occasions...and I'm not seeing an existing tracking task, so I've captured it here:

https://github.com/microsoft/semantic-kernel/issues/5828

Please feel free to add detail.